exotica_core/src/kinematic_tree.cpp

```diff
@@ -1042,14 +1128,17 @@ void KinematicTree::ResetJointLimits()
         {
             auto& ControlledJoint = controlled_joints_map_.at(vars[i]);
             int index = ControlledJoint.lock()->control_id;
+
+            // moveit_core::RobotModel does not have effort limits
+            // TODO: use urdf::Model instead
             controlled_joints_[index].lock()->joint_limits = {model_->getVariableBounds(vars[i]).min_position_, model_->getVariableBounds(vars[i]).max_position_};
+            controlled_joints_[index].lock()->velocity_limit = {model_->getVariableBounds(vars[i]).max_velocity_};
```

Comment:
You can use `acceleration_limited_` here to switch: If not it will write the default value from VariableBounds which I believe is 0?

Reply:
In the line below this I was setting the acceleration to be inf? Do we not want the acceleration limits to default to inf, and then can be set using the setter?

```acceleration_limited_``` does not exist? ```acceleration_bounded_``` is the closest match and that's a bool.
![image](https://user-images.githubusercontent.com/5570173/81860531-cb63c800-955e-11ea-9c18-9891e1a17414.png)

Reply:
My comment is to use the flag `position_bounded_`, `velocity_bounded_`, `acceleration_bounded_` that comes from upstream to decide whether to use their value or use the default inf. If you directly set it to whatever they have as max/min bounds -> that is 0 if there are no limits in the robot model. So, always check that flag from VariableBounds to determine whether to default to inf or use it from there.

Reply:
Ah that makes sense - thanks